ledger/ledger_test.go

```diff
@@ -539,6 +539,164 @@ func TestLedgerSingleTx(t *testing.T) {
+	defer backlogPool.Shutdown()
+
+	protoName := protocol.ConsensusV24
+	genesisInitState, initSecrets := testGenerateInitState(t, protoName)
+	const inMem = true
+	log := logging.TestingLog(t)
+	cfg := config.GetDefaultLocal()
+	cfg.Archival = true
+	l, err := OpenLedger(log, t.Name(), inMem, genesisInitState, cfg)
+	a.NoError(err, "could not open ledger")
+	defer l.Close()
+
+	proto := config.Consensus[protoName]
+	poolAddr := testPoolAddr
+	sinkAddr := testSinkAddr
+
+	initAccounts := genesisInitState.Accounts
+	var addrList []basics.Address
+	for addr := range initAccounts {
+		if addr != poolAddr && addr != sinkAddr {
+			addrList = append(addrList, addr)
+		}
+	}
+
+	correctTxHeader := transactions.Header{
+		Sender:      addrList[0],
+		Fee:         basics.MicroAlgos{Raw: proto.MinTxnFee * 2},
+		FirstValid:  l.Latest() + 1,
+		LastValid:   l.Latest() + 10,
+		GenesisID:   t.Name(),
+		GenesisHash: genesisInitState.GenesisHash,
+	}
+
+	assetParam := basics.AssetParams{
+		Total:    100,
+		UnitName: "unit",
+		Manager:  addrList[0],
+	}
+	correctAssetConfigFields := transactions.AssetConfigTxnFields{
+		AssetParams: assetParam,
+	}
+	correctAssetConfig := transactions.Transaction{
+		Type:                 protocol.AssetConfigTx,
+		Header:               correctTxHeader,
+		AssetConfigTxnFields: correctAssetConfigFields,
+	}
+	correctAssetTransferFields := transactions.AssetTransferTxnFields{
+		AssetAmount:   10,
+		AssetReceiver: addrList[1],
+	}
+	correctAssetTransfer := transactions.Transaction{
+		Type:                   protocol.AssetTransferTx,
+		Header:                 correctTxHeader,
+		AssetTransferTxnFields: correctAssetTransferFields,
+	}
+
+	approvalProgram := []byte("\x02\x20\x01\x01\x22") // int 1
+	clearStateProgram := []byte("\x02")               // empty
+	correctAppCreateFields := transactions.ApplicationCallTxnFields{
+		ApprovalProgram:   approvalProgram,
+		ClearStateProgram: clearStateProgram,
+	}
+	correctAppCreate := transactions.Transaction{
+		Type:                     protocol.ApplicationCallTx,
+		Header:                   correctTxHeader,
+		ApplicationCallTxnFields: correctAppCreateFields,
+	}
+
+	correctAppCallFields := transactions.ApplicationCallTxnFields{
+		OnCompletion: 0,
+	}
+	correctAppCall := transactions.Transaction{
+		Type:                     protocol.ApplicationCallTx,
+		Header:                   correctTxHeader,
+		ApplicationCallTxnFields: correctAppCallFields,
+	}
+
+	var badTx transactions.Transaction
+	var ad transactions.ApplyData
+
+	var assetIdx basics.AssetIndex
+	var appIdx basics.AppIndex
+
+	a.NoError(l.appendUnvalidatedTx(t, initAccounts, initSecrets, correctAssetConfig, ad))
+	assetIdx = 1 // the first txn
+
+	badTx = correctAssetConfig
+	badTx.ConfigAsset = 2
+	err = l.appendUnvalidatedTx(t, initAccounts, initSecrets, badTx, ad)
+	a.Error(err)
+	a.Contains(err.Error(), "asset 2 does not exist or has been delete")
```

Comment:
missing "d" at the end of the string 
`...been delete"` -> `...been deleted"`

Reply:
fixed